docs/adr/0009_versioning_strategy.md

```diff
@@ -0,0 +1,299 @@
+
+## Decision Drivers
+
+* Reproducible, auditable published releases.
+* Maintain per-component autonomy where practical.
+* Support straightforward hotfix and backport workflows for older minor releases.
+* Provide a machine-readable snapshot for documentation and website consumption.
+* Keep CI complexity manageable and deterministic.
+
+## Decision
+
+We adopt a combined, opinionated strategy:
+
+* Versioning: Hybrid approach — each component maintains a local `VERSION` file; the repository also maintains a generated `/VERSIONS.yaml` that mirrors the last published versions of all components (aka snapshot)
+* Branching: Component-scoped persistent release branches — branches follow the pattern `releases/<component>/v<major>.<minor>` and act as maintenance lines for each active minor release. Patches and hotfixes are applied via PRs to these branches.
+* Publishing: An annotated tag `<component>/v<major>.<minor>.<patch>` is the authoritative record of a published artifact.
+
+## SemVer Principles for Root Component
+
+### Normal Case (Automatic)
+
+The `ocm` root component version follows these automated rules for independent sub-component releases:
+
+* **Patch bump (x.y.z → z++)**: triggered by any sub-component patch release
+* **Minor bump (x.y.z → y++)**: triggered by any sub-component feature addition  
+* **Major bump (x.y.z → x++)**: triggered by any sub-component new major release
+
+Every OCM root release represents a reproducible snapshot of all sub-component versions and maintains strict SemVer compliance. The root component version is automatically computed and updated in `/VERSIONS.yaml` when any sub-component publishes a release.
+
+### Edge Case: Coordinated Multi-Component Releases
+
+When multiple sub-components must be released together to solve a single problem:
+
+**Problem:** Automatic root bumping would create multiple OCM releases instead of one logical `ocm`release containing all related changes.
+
+**Solution:** Manual override workflow for coordinated releases:
+
+1. **Sub-component releases with root suppression:**
+
+   Execute the GitHub Action release workflow for each affected sub-component with the "suppress-root-release" flag enabled (flag name to be finalized).
+
+   * Run release workflow for `ocm-cli` with flag: `suppress-root-release = true`
+   * Run release workflow for `ocm-controller` with flag: `suppress-root-release = true`
+   * Each workflow completes its normal release process (tag creation, artifact publishing) but skips the automatic OCM root release trigger.
+
+2. **Manual root release consolidation:**
+
+   After all coordinated sub-component releases are completed, manually trigger the "Create OCM Root Release" workflow:
+
+   * Reads current `/VERSIONS.yaml` state to identify all unreleased changes
```

Comment:
where is this VERSIONS.yaml and how does it look/where is it maintained?

Reply:
then its time for rock-solid CI imho. This will be the heart of our delivery flow and it must "just work" because we will use it weekly, if not daily